modular_zubbers/code/modules/antagonists/bloodsucker/powers/celerity.dm

```diff
@@ -0,0 +1,123 @@
+
+/obj/effect/temp_visual/decoy/celerity
+	duration = 0.25 SECONDS
+	/// The color matrix it should be at spawn
+	var/list/matrix_start = list(1,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,1, 0.5,0,0,0)
+	/// The color matrix it should be by the time it despawns
+	var/list/matrix_end = list(1,0,0,0, 0,0,0,0, 0,0,0,0, 0,0,0,1, 0.2,0,0,0)
+
+/obj/effect/temp_visual/decoy/celerity/Initialize(mapload)
+	. = ..()
+	color = matrix_start
+	animate(src, color = matrix_end, time = duration, easing = EASE_OUT)
+	animate(src, alpha = 0, time = duration, easing = EASE_OUT)
+
+/datum/status_effect/rapid_reflexes
+	id = "Rapid Reflexes"
+	status_type = STATUS_EFFECT_REFRESH
+	duration = 3 SECONDS
+	alert_type = /atom/movable/screen/alert/status_effect/rapid_reflexes
+	tick_interval = 0.2 SECONDS
+	show_duration = TRUE
+
+/datum/status_effect/rapid_reflexes/on_apply()
+	RegisterSignal(owner, COMSIG_ATOM_PRE_BULLET_ACT, PROC_REF(dodge_bullets))
+	owner.add_atom_colour("#d30d0d", TEMPORARY_COLOUR_PRIORITY)
+	owner.add_filter(id, 2, list("type" = "outline", "color" = "#d30d0d", "size" = 1))
+	var/filter = owner.get_filter(id)
+	animate(filter, alpha = 127, time = 1 SECONDS, loop = -1)
+	animate(alpha = 63, time = 2 SECONDS)
+	return TRUE
+
+/datum/status_effect/rapid_reflexes/on_remove()
+	UnregisterSignal(owner, COMSIG_ATOM_PRE_BULLET_ACT)
+	owner.remove_atom_colour(TEMPORARY_COLOUR_PRIORITY, "#d30d0d")
+	owner.remove_filter(id)
+
+/datum/status_effect/rapid_reflexes/proc/dodge_bullets(mob/living/carbon/human/source, obj/projectile/hitting_projectile, def_zone)
+	SIGNAL_HANDLER
+
+	if(HAS_TRAIT(source, TRAIT_INCAPACITATED))
+		return NONE
+	source.visible_message(
+		span_danger("[source] effortlessly dodges [hitting_projectile]!"),
+		span_userdanger("You effortlessly evade [hitting_projectile]!"),
+	)
+	playsound(source, pick('sound/items/weapons/bulletflyby.ogg', 'sound/items/weapons/bulletflyby2.ogg', 'sound/items/weapons/bulletflyby3.ogg'), 75, TRUE)
+	source.add_filter(CELERITY_BLUR, 2, motion_blur_filter(2, 2))
+	addtimer(CALLBACK(source, TYPE_PROC_REF(/datum, remove_filter), CELERITY_BLUR), 0.5 SECONDS)
+	return COMPONENT_BULLET_PIERCED
+
+/atom/movable/screen/alert/status_effect/rapid_reflexes
+	name = "Rapid Reflexes"
+	desc = "Your reflexes are inhuman, enough to dodge bullets."
+	icon_state = "rapid_reflexes"
+
+/datum/action/cooldown/bloodsucker/celerity/DeactivatePower(deactivate_flags)
+	. = ..()
+	if(!.)
+		return
+	var/mob/living/user = owner
+	var/datum/movespeed_modifier/celerity_mod = new()
+	celerity_mod.multiplicative_slowdown = -0.05*level_current
+	owner.remove_movespeed_modifier(celerity_mod, update = TRUE)
+	owner.next_move_modifier /= min(0.05*level_current, 0.5)
+	if(level_current >= CELERITY_FX_LEVEL)
+		UnregisterSignal(owner, COMSIG_MOVABLE_MOVED, PROC_REF(on_movement))
+	user.balloon_alert(user, "the world speeds up.")
+	return TRUE
+
+#undef CELERITY_FX_LEVEL
+#undef CELERITY_DODGE_LEVEL
+#undef CELERITY_BLUR
```

Comment:
missing end of line whitespace ( i hope you're using vscode it automaticallys adds it

Reply:
this is whitespace right
![image](https://github.com/user-attachments/assets/e21b18a2-4ae6-4671-bfae-efd1b482cb36)
cause it's in the actual thing idk why it wouldn't be popping up

Reply:
Wierd, nevermind then